### PR TITLE
Updated Node version to 6.9.5 and also AWS S3 API

### PR DESCRIPTION
**Pull request by saurabh95**

### CODE REVIEW

File: spec/logfileProcessor.spec.js [hunk 1]
```diff
@@ -103,9 +103,7 @@ describe("LogfileProcessor", function () {
                 config: {
                     update: jasmine.createSpy()
                 },
-                S3: {
-                    Client: function (arg) { return S3; }
-                }
+                S3: function (arg) { return S3; }
```
Comment (petetnt) [inline spec/logfileProcessor.spec.js]: Nit: `arg` not used

File: spec/logfileProcessor.spec.js [hunk 2]
```diff
@@ -158,9 +156,7 @@ describe("LogfileProcessor", function () {
                 config: {
                     update: jasmine.createSpy()
                 },
-                S3: {
-                    Client: function (arg) { return s3; }
-                }
+                S3: function (arg) { return s3; }
```
Comment (petetnt) [inline spec/logfileProcessor.spec.js]: Nit: `arg` not used

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - "0.10"
+  - "6.3.1"
```
Comment (petetnt) [inline .travis.yml]: Question: The registry machine has 6.3.1 installed?

Comment (ingorichter) [inline .travis.yml]: Good question. Since I still have no access to that machine, I can't help to verify.

Comment (saurabh95) [inline .travis.yml]: registry machine has node version 0.10.12

File: package.json [hunk 4]
```diff
@@ -27,7 +27,6 @@
     "passport-github": "~0.1.5",
     "semver": "2.x",
     "aws-sdk": "~2.0.0-rc9",
-    "brackets-extensibility": ">0.31.0",
```
Comment (ficristo) [inline package.json]: Why? Wasn't it used before?

**Comment by zaggino** [comment]
btw there's no reason not to use latest LTS release (6.9.5) which contains latest security fixes

**Comment by ingorichter** [comment]
I'm careful when updating from a pre-historic nodejs version to the lastest one. There were changes along the way and I don't know for sure that the registry will work properly after updating it to run on a different and newer node version. Everything might be looking good, but there might also be outdated dependencies that could throw up. I work with nodejs on a daily basis on our product and updating caused some headaches for us here and there.

**Comment by saurabh95** [comment]
@zaggino I will try upgrading node version to 6.9.5
@ingorichter I have tested the downloadStats part which is run as cron job, it is working fine with my changes and views are also working fine(tested it in local)
But there might be some issues in the extension upload or extension update so for testing that I will expose the https://brackets-registry.aboutweb.com/ with a different port and try uploading/updating test extension, if everything works fine, then I will update the server code :)

**Comment by swmitra** [comment]
Nobody has permission to the served location in aboutweb server. It seems the code running in the server is almost 2 years old and doesn't include commits by @MarcelGerber and @ingorichter. Using nvm we have got in both the node versions into the server but can't use the newer version unless we upgrade the registry server code.

**Comment by nethip** [comment]
@ingorichter Thanks for pointing the versioning problem out. We will extensively test it before updating with server with the new NodeJS version. Can you tell us if there are any unit tests for the server code?

**Comment by swmitra** [comment]
Merging the PR. We will pull latest master to the registry server once we get permission to restart the service.

**Pull request merged**